usr/lib/cca_stdll/cca_specific.c

```diff
@@ -773,12 +775,21 @@ CK_RV token_create_keypair_object(TEMPLATE * tmpl, CK_ULONG tok_len,
     }
     template_update_attribute(tmpl, pub_exp);
 
+    return CKR_OK;
+}
+
+static CK_RV add_ibm_opaque_to_template(TEMPLATE *tmpl,
+					CK_BYTE *tok, CK_ULONG tok_len)
+{
+    CK_RV rv;
+    CK_ATTRIBUTE *opaque;
+
     /* Add the opaque key object to the template */
-    if ((rv = build_attribute(CKA_IBM_OPAQUE, tok, tok_len, &opaque_key))) {
-        TRACE_DEVEL("build_attribute for opaque key failed. rv=0x%lx\n", rv);
+    if ((rv = build_attribute(CKA_IBM_OPAQUE, tok, tok_len, &opaque))) {
+        TRACE_DEVEL("build_attribute for opaque attribute failed. rv=0x%lx\n", rv);
         return rv;
     }
-    template_update_attribute(tmpl, opaque_key);
+    template_update_attribute(tmpl, opaque);
```

Comment:
Use build_update_attribute() instead of this whole function? 

If you choose to use it the way you have it, then please add error checking for template_update_attribute as it is done in the commit in my PR: https://github.com/opencryptoki/opencryptoki/pull/336/commits/80e624f94615aa80ba751596f80f160791555cbc#diff-a14e5f28856cbf95c58b0212766dd738551a826b75496d26aa711ac06c9ac14aR830

Reply:
Ingo, you are right. I'll update the pull request and use the build_update_attribute() function. :-)

Reply:
pull request is updated - please note I added a whitespace commit in front of the patch because my editor
permanently wants to fix these issues and is driving me mad with mixing my code changes with whitespace updates.